### DISCUSSION TITLE
Mark the current topic which is visible on the screen in the side menu on nicegui.io/documentation

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#1592] [Ideas / Feature Requests]
Could [nicegui](https://nicegui.io/documentation#styling) make  such effect like this ?
![微信截图_20230911015710](https://github.com/zauberzeug/nicegui/assets/25096045/dd00a162-f53f-4109-9643-042d1176a137)

**Comment by rodja** [comment]
Sure. We have done so in the NiceGUI documentation:

<img width="649" alt="Screenshot 2023-09-11 at 05 25 40" src="https://github.com/zauberzeug/nicegui/assets/131391/efe0c623-4014-4e49-b19d-7cae9e1fad4b">


Here is a minimal example of using link_targets: https://nicegui.io/documentation/link#navigate_on_large_pages

**Reply by v479038280** [reply]
When the page content  scroll ,the drawer will also scroll to the link 。Nicegui need to scroll manually to find the correspond link

**Reply by rodja** [reply]
I'm not 100% sure what you mean. You want the navigation to stay fix? That can be achieved with tailwind props or css.

**Reply by v479038280** [reply]
I made two Gif to show the difference.
here is nicegui . when scrolling ,the left_drawer stays the same.
![GIF 2023-9-11 12-13-52](https://github.com/zauberzeug/nicegui/assets/25096045/6ece42aa-60fc-46e2-a047-a7b55b9fa778)



and this one from justpy document, the right_drawer changes when scrolling,
![GIF 2023-9-11 12-20-03](https://github.com/zauberzeug/nicegui/assets/25096045/403c1b77-89cc-4e66-ba58-47697c396c09)

**Comment by rodja** [comment]
Ah, I see. You want to highlight the element in the side bar according to the scroll position in the main document. The side bar could be a [ui.refreshable with parameter](https://nicegui.io/documentation/refreshable#refreshable_ui_with_parameters) which defines what element should be highlighted. Then use a `ui.scroll_area` for main content and trigger a refresh depending on the scroll position.

Not the easiest thing to implement, but possible I guess.

**Reply by v479038280** [reply]
Not just highlight the element,but make the current element in the proper view, I think nice gui.io. could integrate such effect, It would be more convenient

**Reply by rodja** [reply]
Ah! You do not want to implement it in your own app but in the official documentation. Ok. Anyone interested in creating a pull request for this?

**Reply by v479038280** [reply]
I'm not familar with JS ,It may need to use JS to dynamically scroll the page and drawer.

**Comment by eunfngswr** [comment]
I too hope that this will be officially supported rather than implemented by myself.

**Reply by natankeddem** [reply]
The Quasar site implements something similar with QList and QItem...
https://quasar.dev/vue-components/list-and-list-items
I am not sure if that is something that might be leveraged here?

**Comment by falkoschindler** [comment]
I could imagine something like a `ui.table_of_contents` element that (optionally) highlights the current section title. But for it to work we need a counterpart like `ui.heading` or `ui.section` which registers the actual content of the table of contents. Alternatively we simply collect all h1, h2, etc. tags. But this would be rather implicit and too close to good-old HTML for my taste.

A `ui.section` element (Hi, LaTeX! 👋) raises the question about it's default style. We could use something from [Quasar's Typography classes](https://quasar.dev/style/typography#headings) though.
